core/trace/src/main/scala/org/typelevel/otel4s/trace/KindTransformer.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.trace
+
+import cats.Applicative
+import cats.Functor
+import cats.Monad
+import cats.data.EitherT
+import cats.data.IorT
+import cats.data.Kleisli
+import cats.data.Nested
+import cats.data.OptionT
+import cats.data.StateT
+import cats.effect.MonadCancelThrow
+import cats.effect.Resource
+import cats.implicits.toFunctorOps
+import cats.syntax.all._
+import cats.~>
+
+/** A utility for transforming the higher-kinded type `F` to another
+  * higher-kinded type `G`.
+  */
+trait KindTransformer[F[_], G[_]] {
```

Comment:
Perhaps the same approach can be used in the `metrics` module too. If so, the `KindTransformer` can be moved to the `common` module.

Reply:
incidentally, that would also allow adding the convenience `def mapK[G[_]](implicit kt: KindTransformer[F, G]) = mapK(kt.liftK)` to `InstrumentMeta`... although that might break things with `Tracer.Meta`

Reply:
unless we're just looking for the convenience of having it implicitly available and not needing to type `<Type>.liftK` at use sites of `mapK`, the only reason we'd *need* `KindTransformer` is if there are functions/methods from `F[A]` to `F[A]` that we need to transform, or if there's a natural transformation we need to transform. I haven't seen any of the latter, and I'm not immediately finding any cases of the former.

on the other hand, nothing about `KindTransformer` is specific to `trace` per se, and there's also no particular reason *not* to have it in `common`, even if it ends up being of no benefit to `metrics`